.github/workflows/kiali-ci.yml

```diff
@@ -1,23 +1,41 @@
-name: Kiali Go CI
+name: Kiali CI
 
 on:
+  # Run on master and release branches
   push:
-    branches: [master]
+    branches: [master, "^v\\d+\\.\\d+$"]
   pull_request:
-    branches: [master]
+    branches: [master, "^v\\d+\\.\\d+$"]
+
+env:
+  # This workflow can run on either a PR or release. GITHUB_BASE_REF is only
+  # set when this workflow is run against a PR. When run against a PR, we want
+  # to use the branch we are running against: GITHUB_BASE_REF. When run
+  # as a release, we want to use the current branch ref: GITHUB_REF_NAME.
+  TARGET_BRANCH: ${{ github.base_ref || github.ref_name }}
 
 jobs:
-  build:
-    name: Build
+  build_backend:
```

Comment:
I'm wondering if we can turn build_backend and build_frontend jobs into reusable workflows (https://docs.github.com/en/actions/using-workflows/reusing-workflows). 

Something similar to the integration one. 

This is because in the release pipeline I'm working on I have very similar steps, that, can be the same I guess.

With this approach, we will have reusable workflows and both release and ci pipelines will use them.

Reply:
It looks like just adding `workflow_call:` will make this reusable? I can include that in this change.

Reply:
It is just that, but what I wanted to do is to split these jobs in 3 files like:
https://github.com/leandroberetta/kiali/tree/release-pipeline/.github/workflows

So, with these common jobs, both pipelines can use them.

The test worked.

Reply:
Ah so you can't reference individual jobs within a workflow it looks like? Well if it's alright with you, I'll leave making the workflow reusable up to you in your PR. This change was mostly about getting an e2e env setup in CI and having something use it to run tests on every PR.